server/src/prefect_server/api/flows.py

```diff
@@ -83,8 +83,17 @@
         f = prefect.serialization.flow.FlowSchema().load(copy.deepcopy(serialized_flow))
     except Exception as exc:
         raise ValueError(f"Invalid flow: {exc}")
-    if f.schedule and any([p.required for p in f.parameters()]):
-        raise ValueError("Can not schedule a flow that has required parameters.")
+    required_parameters = [p for p in f.parameters() if p.required]
+    if f.schedule is not None and required_parameters:
+        required_names = {p.name for p in required_parameters}
+        if not all(
+            [
+                required_names == set(c.parameter_defaults.keys())
```

Comment:
I think this might need some set intersection to be complete.

For example, if I have two clocks - one that provides a default for a, and one that provides defaults for a and b, then I think `required_names == {'a'}` but `set(c.parameter_defaults.keys()) == {'a', 'b'}`

Reply:
That wouldn't work though -- each unique scheduled clock event needs values for all required parameters.

Reply:
Right, but in addition to supplying a default for each required parameter, a clock could include defaults for non-required parameters as well. Therefore as long as the clock supplies *as least* the required keys, it's good to go. The current check says it must supply *only* the required keys.

Reply:
Ahhhh sorry I understand

Reply:
@jlowin updated